static/js/components/OrderSummaryCard.js

```diff
@@ -0,0 +1,165 @@
+// @flow
+/* global SETTINGS: false */
+import React from "react"
+import { Button } from "reactstrap"
+import { formatLocalePrice, isSuccessResponse } from "../lib/util"
+import ApplyCouponForm from "./forms/ApplyCouponForm"
+import type { Discount } from "../flow/cartTypes"
+
+type Props = {
+  totalPrice: number,
+  orderFulfilled: boolean,
+  discountedPrice: number,
+  discounts: Array<Discount>,
+  addDiscount?: Function,
+  clearDiscount?: Function,
+  discountCode: string,
+  discountCodeIsBad: boolean,
+  cardTitle?: string
+}
+
+export class OrderSummaryCard extends React.Component<Props> {
+  renderAppliedCoupons() {
+    const { discounts, clearDiscount, orderFulfilled } = this.props
+
+    if (discounts === null || discounts.length === 0) {
+      return null
+    }
+
+    let discountAmountText = null
+    const discountAmount = Number(discounts[0].amount)
```

Comment:
discounts[x] has a "redeemed_discount" key - as is this renders as NaN (and also has problems on line 32 and 58)